equal.go

```diff
@@ -18,58 +18,25 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package config_test
+package config
 
 import (
+	"bytes"
 	"fmt"
-	"log"
 
-	"go.uber.org/config"
+	yaml "gopkg.in/yaml.v2"
 )
 
-func ExampleNewProviderGroup() {
-	base, err := config.NewYAMLProviderFromBytes([]byte(`
-config:
-  name: fx
-  pool: development
-  ports:
-  - 80
-  - 8080
-`))
-
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	prod, err := config.NewYAMLProviderFromBytes([]byte(`
-config:
-  pool: production
-  ports:
-  - 443
-`))
-
+// areSameYAML checks whether two values represent the same YAML data.
+func areSameYAML(left, right interface{}) (bool, error) {
+	l, err := yaml.Marshal(left)
 	if err != nil {
-		log.Fatal(err)
+		// unreachable with YAML provider, but possible with third-party provider
```

Comment:
How is a third-party provider relevant here? Aren't we just calling `gopkg.in/yaml.v2.Marshal` on a configuration type without a provider?

Reply:
I'll clarify the docs here. We only call this function from `NewValue`, where the provider is user-supplied. Since we can't guarantee that the user-supplied provider is the one from this package, we can't absolutely guarantee that the value is representable in YAML.